### PR TITLE
perf(log): encode objects only when logged

### LINKED ISSUE #480: logging message performance needs to be improved
### What is not working as expected?

```go
logger.Errorf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject.String(), serverResult.RevocationMethod, serverResult.Server, serverResult.Error)
```

Code like this example should be optimized to avoid rendering the string before actually logging it.
```go
logger.Errorf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject, serverResult.RevocationMethod, serverResult.Server, serverResult.Error)
```

### What did you expect to happen?

nil

### How can we reproduce it?

nil

### Describe your environment

source code

### What is the version of your notation-go Library?

v1.3.0-rc.1

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Fix:
- replaced `.String()` with the `%v` format to avoid rendering the string before actually logging it.

Resolves #480

### CODE REVIEW

**Review by Two-Hearts — CHANGES_REQUESTED** [review]
Please fix all occurrences in the library.

Nvm, looks like both files are covered.

File: verifier/verifier.go [hunk 1]
```diff
@@ -812,7 +812,7 @@ func revocationFinalResult(certResults []*revocationresult.CertRevocationResult,
 		certResult := certResults[i]
 		if certResult.RevocationMethod == revocationresult.RevocationMethodOCSPFallbackCRL {
 			// log the fallback warning
-			logger.Warnf("OCSP check failed with unknown error and fallback to CRL check for certificate #%d in chain with subject %v", (i + 1), cert.Subject.String())
+			logger.Warnf("OCSP check failed with unknown error and fallback to CRL check for certificate #%d in chain with subject %v", (i + 1), cert.Subject)
```
Comment (Two-Hearts) [inline verifier/verifier.go:815]:
```suggestion
			logger.Warnf("OCSP check failed with unknown error and fallback to CRL check for certificate #%d in chain with subject %q", (i + 1), cert.Subject)
```

File: verifier/verifier.go [hunk 2]
```diff
@@ -821,10 +821,10 @@ func revocationFinalResult(certResults []*revocationresult.CertRevocationResult,
 					// when the final revocation method is OCSPFallbackCRL,
 					// the OCSP server results should not be logged as an error
 					// since the CRL revocation check can succeed.
-					logger.Debugf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject.String(), revocationresult.RevocationMethodOCSP, serverResult.Server, serverResult.Error)
+					logger.Debugf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject, revocationresult.RevocationMethodOCSP, serverResult.Server, serverResult.Error)
```
Comment (Two-Hearts) [inline verifier/verifier.go:824]:
```suggestion
					logger.Debugf("Certificate #%d in chain with subject %q encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject, revocationresult.RevocationMethodOCSP, serverResult.Server, serverResult.Error)
```

File: verifier/verifier.go [hunk 3]
```diff
@@ -821,10 +821,10 @@ func revocationFinalResult(certResults []*revocationresult.CertRevocationResult,
...
 					continue
 				}
-				logger.Errorf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject.String(), serverResult.RevocationMethod, serverResult.Server, serverResult.Error)
+				logger.Errorf("Certificate #%d in chain with subject %v encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject, serverResult.RevocationMethod, serverResult.Server, serverResult.Error)
```
Comment (Two-Hearts) [inline verifier/verifier.go:827]:
```suggestion
				logger.Errorf("Certificate #%d in chain with subject %q encountered an error for revocation method %s at URL %q: %v", (i + 1), cert.Subject, serverResult.RevocationMethod, serverResult.Server, serverResult.Error)
```

File: verifier/verifier.go [hunk 4]
```diff
@@ -840,7 +840,7 @@ func revocationFinalResult(certResults []*revocationresult.CertRevocationResult,
 		}
 
 		if i < len(certResults)-1 && certResult.Result == revocationresult.ResultNonRevokable {
-			logger.Warnf("Certificate #%d in the chain with subject %v neither has an OCSP nor a CRL revocation method.", (i + 1), cert.Subject.String())
+			logger.Warnf("Certificate #%d in the chain with subject %v neither has an OCSP nor a CRL revocation method.", (i + 1), cert.Subject)
```
Comment (Two-Hearts) [inline verifier/verifier.go:843]:
```suggestion
			logger.Warnf("Certificate #%d in the chain with subject %q neither has an OCSP nor a CRL revocation method.", (i + 1), cert.Subject)
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: verifier/verifier.go (hunk 2 above)
Comment (shizhMSFT) [reply verifier/verifier.go:824]: This PR aims for perf improvements. We can create another PR to fix the content to be logged. The same comment applies to other occurrences.

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: verifier/verifier.go (hunk 2 above)
Comment (JeyJeyGao) [reply verifier/verifier.go:824]: Sure. I can create another PR to update the format.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**